src/Chapter2.hs

```diff
@@ -348,9 +360,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l = take half l
+    where
+      half = (div (length l) 2)
```

Comment:
Great usage of the `where` construction!

You can omit brackets here:

```suggestion
      half = div (length l) 2
```

Reply:
Thanks! I put too many brackets in general I see